### PR TITLE
kubebuilder v3

### PR DESCRIPTION
**Pull request by zoetrope**

### CODE REVIEW

File: codes/tenant/config/manager/manager.yaml [hunk 1]
```diff
@@ -22,33 +22,37 @@ spec:
...
-          periodSeconds: 3
+            path: /readyz
+            port: 8081
+          initialDelaySeconds: 15
+          periodSeconds: 20
         readinessProbe:
           httpGet:
-            path: /readyz
-            port: 9090
+            path: /healthz
```
Comment (ymmt2005) [inline codes/tenant/config/manager/manager.yaml:46]: paths seem wrong.

File: codes/tenant/controllers/tenant_controller.go [hunk 2]
```diff
@@ -25,19 +25,27 @@ type TenantReconciler struct {
 	Log      logr.Logger
 	Scheme   *runtime.Scheme
 	Recorder record.EventRecorder
-	stopCh   <-chan struct{}
 }
 
 //! [rbac]
 // +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=multitenancy.example.com,resources=tenants/finalizers,verbs=update
```
Comment (ymmt2005) [inline codes/tenant/controllers/tenant_controller.go:33]: I cannot find `finalizers` subresource support for CRDs.
https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#subresources

Comment (zoetrope) [reply codes/tenant/controllers/tenant_controller.go:33]: It seems to be used in the `OwnerReferencesPermissionEnforcement` admission controller.
(However, it's not enabled by default.)

https://github.com/kubernetes-sigs/kubebuilder/pull/1688
https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement
https://github.com/kubernetes/kubernetes/blob/87984d84d18c7816835e5efedebb17ed13aa2509/plugin/pkg/admission/gc/gc_admission.go

File: codes/tenant/config/manager/manager.yaml (hunk 1 above)
Comment (zoetrope) [reply codes/tenant/config/manager/manager.yaml:46]: This may be a bug in kubebuilder.

File: codes/tenant/controllers/tenant_controller.go (hunk 2 above)
Comment (ymmt2005) [reply codes/tenant/controllers/tenant_controller.go:33]: Understood.

**Pull request merged**